### PR TITLE
Make Captures grant experience based on targets cost

### PR DESCRIPTION
**Pull request by PunkPun**
Killing a unit grants experience based of 1% of the cost

Capturing will now give 2% of the captured cost, with the exception of RA mechanic and TD engineer capturing husks, as you already get 1% from the initial kill.

### CODE REVIEW

File: OpenRA.Mods.Common/Activities/CaptureActor.cs [hunk 1]
```diff
@@ -125,7 +125,7 @@ void DoCapture(Actor self, Captures captures)
 					t.OnCapture(enterActor, self, oldOwner, self.Owner, captures.Info.CaptureTypes);
 
 				if (self.Owner.RelationshipWith(oldOwner).HasRelationship(captures.Info.PlayerExperienceRelationships))
-					self.Owner.PlayerActor.TraitOrDefault<PlayerExperience>()?.GiveExperience(captures.Info.PlayerExperience);
+					self.Owner.PlayerActor.TraitOrDefault<PlayerExperience>()?.GiveExperience(captures.Info.PlayerExperience + captures.Info.PlayerExperiencePercentage * (enterActor.Info.TraitInfo<ValuedInfo>()?.Cost ?? 0) / 100);
```
Comment (abcdefg30) [inline OpenRA.Mods.Common/Activities/CaptureActor.cs]: This should probably use `TraitInfoOrDefault`?

Comment (PunkPun) [reply OpenRA.Mods.Common/Activities/CaptureActor.cs]: Yes, fixed

File: mods/cnc/rules/infantry.yaml [hunk 2]
```diff
@@ -211,10 +211,13 @@ E6:
 	Captures@SABOTAGE:
 		CaptureTypes: building-sabotage
 		SabotageThreshold: 55
-		PlayerExperience: 50
+		PlayerExperience: 5
```
Comment (Mailaender) [inline mods/cnc/rules/infantry.yaml]:
```suggestion
		PlayerExperiencePercentage: 5
```
?

Comment (PunkPun) [reply mods/cnc/rules/infantry.yaml]: Oh my bad. I though sabotage was only the damage, and the capturing part was below

**Review by Mailaender — DISMISSED** [review]

**Comment by MustaphaTR** [comment]
Repairing husks currently do not give any experience in either mod with the function because Husks do not have a cost, actual units do.

**Comment by PunkPun** [comment]
Hmm, how would you fix that. I don't like the idea of manually setting XP per husk, as a massive maintenance overhead. Is there a way to check the cost of a unit before transforming into it?

**Pull request closed without merging**